api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -76,6 +76,22 @@
    */
   String TARGET_FILE_SIZE_BYTES = "target-file-size-bytes";
 
+  /**
+   * Determines if the data rewrite action should also remove non-global deletes associated with the data files.
+   * By enabling this option, any data filter specified through {@link #filter(Expression)} will be converted to
+   * an inclusive partition filter based on all the historical partition specs of the table.
+   */
+  String REMOVE_PARTITION_DELETES = "remove-partition-deletes";
+  boolean REMOVE_PARTITION_DELETES_DEFAULT = false;
+
+  /**
+   * Determines if the data rewrite action should also remove global deletes.
+   * When enabling this option, specify a data filter would result in {@link IllegalArgumentException}
+   * because a full table scan planning must be performed to safely remove global deletes.
```

Comment:
per https://iceberg.apache.org/#spec/#equality-delete-files, the equality delete file may optionally contain columns of the deleted row.  Could those optional columns along with the equality columns serve as a hint to locate the rows to delete, IFF the optional and equality columns happen to intersect with partition columns ?

@puneetzaroo